include/backup_restore.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include "types.hpp"
+
+#include <nlohmann/json.hpp>
+
+#include <tuple>
+
+namespace vpd
+{
+/**
+ * @brief class to implement backup and restore VPD.
+ *
+ */
+class BackupRestore
+{
+  public:
+    // delete functions
+    BackupRestore() = delete;
+    BackupRestore(const BackupRestore&) = delete;
+    BackupRestore& operator=(const BackupRestore&) = delete;
+    BackupRestore(BackupRestore&&) = delete;
+    BackupRestore& operator=(BackupRestore&&) = delete;
+
+    /**
+     * @brief Constructor.
+     *
+     * @param[in] l_parsedSystemJson - Parsed system config JSON.
+     *
+     */
+    BackupRestore(const nlohmann::json& i_parsedSystemJson);
+
+    /**
+     * @brief Defaul destructor.
+     */
+    ~BackupRestore() = default;
+
+    /**
+     * @brief An API to backup and restore VPD.
+     *
+     * Note: This API works on the keywords declared in the backup and restore
+     * config JSON. Restore or backup action could be triggered for each
+     * keyword, based on the keyvalue present in the source and destination
+     * keyword.
+     *
+     * Restore source keyword value with destination keyword value,
+     * when source keyword has default value but
+     * destination has non default value.
+     *
+     * Backup the source keyword value on destination,
```

Comment:
nit: to the destination's keywords value

Reply:
Done